extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationService.java

```diff
@@ -28,10 +38,20 @@ public class RegistrationService {
 
     private final Monitor monitor;
     private final ParticipantStore participantStore;
+    private final CredentialsVerifier credentialsVerifier;
+    private final StateMachine stateMachine;
```

Comment:
I would extract this state machine to a separate class. Now it seems that Registration service has multiple responsibilities - starts a state machine and manages the participants. 
It's not selfexplanatory why RegistrationServcie starts() starts a state machine. 
Wy could have another object, eg. ParticipantsStateMachine that could be started in the RegistrationServiceExtension and then advancing the states would be decoupled from the RegistrationService which would remain light (as soon as we have simple in-memory store).
I tried it myself to see how it would look like and pushed to a branch: https://github.com/agera-edc/RegistrationService/pull/10 
Feel free to take a look

Reply:
Thanks, great idea. I've taken it, and named the new class ParticipantsManager (for consistency with how such classes are named in EDC)